Week2/MAKEME.md

```diff
@@ -18,36 +18,110 @@ implement them:
 
 *Read how to [use and escape query values](https://github.com/mysqljs/mysql#escaping-query-values)*
 
-```js
-class TodoModel {
-
-    // ...
+// This is the connector (also known as driver)
+// that we can use to connect to a MySQL process
+// and access its databases.
+const mysql = require('mysql');
 
-    create(description, callback) {
-        // Write code and query to create a new TODO item
+class TodoModel {
+    constructor(dbConnection) {
+        this.dbConnection = dbConnection;
     }
 
-    update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+    // Loads all the TODOs in the database
+    load(callback) {
+        const selectTodoItems = "SELECT * FROM todo_items";
+        this.dbConnection.query(selectTodoItems, function (err, results, fields) {
+            if (err) {
+                callback(err);
+                return;
+            }
+
+            callback(null, results);
+        });
     }
 
-    delete(id, callback) {
-        // Write code and query to delete an existing TODO item
-    }
 
-    tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
-    }
-        
-    untagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query remove a tag from a TODO item
-    }
 
-    markCompleted(todoItemId, callback) {
-        // Write code to mark a TODO item as completed
-    }
+
+    function create(id, description, callback) {
+    // Write code and query to create a new TODO item
+    var query = dbConnection.query('INSERT INTO todo_items (text, user_id) VALUES ( ' + description + ' , ' + id + ' )');
+    query.on('Result', function (row) {
+        callback(null, row.id)
+    });
 }
-```
+
+
+
+function update(id, description, callback) {
+    // Write code and query to update and existing TODO item
+    var query = dbConnection.query('UPDATE todo_items SET text = ' + description + 'Where user_id = ' + id);
+    query.on('Result', function (row) {
+        callback(null, row.id)
+    });
+}
+
+
+
+function delete (id, callback) {
+    // Write code and query to delete an existing TODO item
+    var query = dbConnection.query('DELETE FROM todo_items WHERE user_id = ' + id);
+    query.on('Result', function (row) {
+        callback(null, row.id)
+    });
+}
+
+function tagTodoItem(todoItemId, tagId, callback) {
+    // Write code and query add a tag to a TODO item
+    var query = dbConnection.query('INSERT INTO  todo_item_tag(todoItemId, tagId) VALUES ( ' + todoItemId + ' , ' + tagId + ' )');
```

Comment:
:+1: 

This doesn't handle the case where a new tag is being created, though - ie. that tag has not been used before and doesn't already exist.